### ISSUE TITLE
substract operation ignore brackets

### ISSUE THREAD

**Issue by mkupriichuk** [juanluispaz/ts-sql-query#53]
Hello, we have two ```period```

allPeriodQuery:
```
const allPeriodQuery = connection
      .selectFrom(agOrderTable)
      .where(
        agOrderTable.date
          .greaterOrEquals(previousPeriod.from)
          .and(agOrderTable.date.lessOrEquals(currentPeriod.to))
      )
      .select({
        difference: connection.sum(agOrderTable[metricField])
      })
     .forUseInQueryAs('allPeriodQuery');
```

And currentPeriodQuery
```
    const currentPeriodQuery = connection
      .selectFrom(agOrderTable)
      .where(
        agOrderTable.date
          .greaterOrEquals(current.from)
          .and(agOrderTable.date.lessOrEquals(currentPeriod.to))
      )
      .select({
        difference: connection.sum(agOrderTable[metricField])
      })
      .forUseInQueryAs('allPeriodQuery');
```

Next, we need to calculate
```current - previous```

We do not have a ```previous```, but it is equal to ```allPeriod - currentPeriod```

So we do next:
```currentPeriodQuery.difference.substract(allPeriodQuery.difference.substract(currentPeriodQuery.difference))```

As a result, we get
```currentPeriodQuery.difference - allPeriodQuery.difference - currentPeriodQuery.difference```

But I think it should be with brackets ```currentPeriodQuery.difference - (allPeriodQuery.difference - currentPeriodQuery.difference)```

**Comment by juanluispaz** [comment]
Fixed in ts-sql-query 1.40.0, let me know if it works well for you

**Comment by mkupriichuk** [comment]
> Fixed in ts-sql-query 1.40.0, let me know if it works well for you

yes, brackets correct now. And one more request:

Can you add ```.nullIfValue(value)``` method? Its need for avoid divide by zero
At the moment we use ```connection.fragmentWithType``` to do this:

```
.divide(connection.fragmentWithType('int', 'required').sql`IFNULL(NULLIF(${previousPeriodCount}, 0), 1)`)
```

It would be nice to be able use some next
```
.divide(previousPeriodCount.nullIfValue(0).valueIfNull(1))
```

**Comment by juanluispaz** [comment]
Implemented `nullIfValue` in ts-sql-query 1.41.0 already published; can you test it?, please

**Comment by mkupriichuk** [comment]
Yes, works nice. Thanks
